recipe/meta.yaml

```
@@ -1,54 +1,56 @@
 {% set name = "tensorflow-probability" %}
-{% set version = "0.24.0" %}
+{% set version = "0.25.0" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://github.com/tensorflow/probability/archive/v{{ version }}.tar.gz
-  sha256: 3d75418cef09ea357ee879347133ab784fe4637a5ba251a8e06ef930dd970a3e
-  patches:
-    - 0001-always-build-release.patch
+  url: https://pypi.org/packages/py2.py3/t/tensorflow-probability/tensorflow_probability-{{ version }}-py2.py3-none-any.whl
+  sha256: f3f4d6431656c0122906888afe1b67b4400e82bd7f254b45b92e6c5b84ea8e3e
 
 build:
   noarch: python
   number: 0
-  script:
-    - bazel build :pip_pkg
-    - mkdir tmp-pkg
-    - ./bazel-bin/pip_pkg tmp-pkg
-    - "{{ PYTHON }} -m pip install --upgrade --no-deps -vv tmp-pkg/*.whl"
+  script: {{ PYTHON }} -m pip install -vv --no-deps --no-build-isolation tensorflow_probability-{{ version }}-py2.py3-none-any.whl
+
 requirements:
-  build:
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-    - bazel
   host:
-    - python >=3.9
+    - python {{ python_min }}
     - pip
-    - absl-py
+    - setuptools
   run:
-    - python >=3.9
-    - tensorflow-base >=2.16.0, <2.17.0
-    - tf-keras >=2.16  # required as TFP only supports Keras2, see  https://github.com/tensorflow/probability/blob/v0.24.0/setup.py#L51
-    - jax >=0.4.25, <0.5.0
-    - numpy >=1.13.3
+    - python >={{ python_min }}
     - absl-py
     - six >=1.10.0
+    - numpy >=1.13.3
     - decorator
     - cloudpickle >=1.3
     - gast >=0.3.2
     - dm-tree
+  run_constrained:
+    # Optional dependencies: tensorflow and tf-keras / tensorflow-datasets / jax and jaxlib (without version constraints)
+    - tensorflow >=2.16.0
+    # Required with TensorFlow as this package only supports Keras 2
+    - tf-keras >=2.16
+    - tensorflow-datasets >=2.2.0
 
 test:
+  requires:
```

```suggestion
    - jax >=0.4.35, <0.5.0

test:
  requires:
    - tf-keras
    - tensorflow-base
    - jax
```

Adding JAX to the constraints (relaxed) and tf with keras to the test run requirements

---

Where was the source of this restraint? I don't see this within the code as well.

---

Committed in https://github.com/conda-forge/tensorflow-probability-feedstock/pull/56/commits/e769a74b9a39a2ffe34728e0b752d2154b579b62.

---

Not quite I think. Does 0.4.1 (do you mean 0.4.0) mean the lowest in the 0.4.x series?
If so, JAX [did not](https://docs.jax.dev/en/latest/jep/25516-effver.html#non-semantic-versioning-status-quo) use semantic versioning. There can be breaking changes between 0.4.35 and 0.4.34 (and there are in general). They switched to effortbased versioning. Nice. Still doesn't tell anything about compatibility and therefore doesn't help us here a lot.
I therefore think it's best to keep the lower bound. Or where should we set it? I don't think it's a big deal, as it won't break peoples code. It will only prevent them from upgrading _without upgrading JAX_. But that's it.

For 0.5.0, feel free to test the whole suite, (that would be ideal actualy), if we're reasonably sure that we can lift this restriction, then let's do that. If TFP in pypi doesn't restrict it, why should we here on conda

---

It also seems that JAX 0.5.0 doesn't exist in conda-forge yet, so we don't know.